pkg/cmd/component/cmd/ui/ui.go

```diff
@@ -403,3 +403,81 @@ func AddEnvironmentName() string {
 	handleError(err)
 	return strings.TrimSpace(environmentName)
 }
+func ApplicationOutputPath() string {
+	var path string
+	prompt := &survey.Input{
+		Message: "Provide a path to application folder",
+		Help:    "This is the path where application configuration is stored locally before you push it to the repository GitRepoURL",
+	}
+	err := survey.AskOne(prompt, &path, validateAppPath())
+	handleError(err)
+	return strings.TrimSpace(path)
+}
+
+func validateAppPath() survey.Validator {
+	return func(input interface{}) error {
+		return validateAppFolder(input)
+	}
+}
+
+func validateAppFolder(input interface{}) error {
```

Comment:
If you specify the current folder by typing
.
and returning, should we allow it?

For example, the current folder is /Users/jainam/myapps/app1 and it is empty.
Then you run `kam init` from this folder, should we allow it?  Therefore app1 is actually the application folder.
If we allow this, then we should verify that the folder is indeed empty and then it is safe to add the .git folder.  However, if it is not empty and the components folder DOES NOT exist, then should we add the .git folder, or should we fail with an error message?  

Another way to view it is that in the terminal, you change directory (cd) to one of the parent folders, so for example, the PWD is /Users/jainam/myapps

Then you run `kam init` and for the path to the application, you specify ./app1 as the application path.

This scenario should be allowed.

Reply:
`
l := ListFiles(u)
			if len(l) != 0 {
				exists, _ = ioutils.IsExisting(ioutils.NewFilesystem(), filepath.Join(u, "components"))
				if !exists {
					err := fmt.Errorf("the given Path : %s is not a correct path for an application ", u)
					return (err)
				}

			}`
if you check this. It takes care : when folder exists and if there are files present then it should have component in the sub directory. So I think both the scenarios are getting covered. 

If this is the case /Users/jainam/myapps/app1  then it will allow the user to create a .git file if .git file is not present.